internal/controller/status.go

```diff
@@ -117,41 +127,58 @@ func getIdleState(
 		return metav1.Time{}, false
 	}
 	idleSince := cr.Status.IdleSince
-	idle := false
 
+	cpuIdle := Unknown
+	var cpuUsage *resource.Quantity
 	metrics, err := metrics(ctx, r.MetricsClient, cr)
 	if err != nil {
-		log.Info("Metrics returned error", "error", err)
-		return metav1.Time{}, false
+		log.Info("Metrics returned error when checking idleness", "error", err)
+	} else {
+		cpuUsage = metrics.Cpu()
 	}
-
-	cpuUsage := metrics.Cpu()
-	if cpuUsage != nil && cpuUsage.Cmp(cpuUsageIdlenessThreshold) == -1 {
-		if cr.Status.IdleSince.IsZero() {
-			log.Info(
-				"the session was found to be idle",
-				"cpu usage milicores",
-				cpuUsage.MilliValue(),
-				"idle threshold milicores",
-				cpuUsageIdlenessThreshold.MilliValue(),
-			)
+	if cpuUsage != nil {
+		if cpuUsage.Cmp(cpuUsageIdlenessThreshold) == -1 {
+			cpuIdle = Idle
+		} else {
+			cpuIdle = NotIdle
 		}
-		idle = true
-		// check last request time before setting session to idle
-		rt, err := lastRequestTime(cr)
-		if err == nil && (idleSince.IsZero() || idleSince.Time.Before(rt)) {
-			log.Info("the last request to the session happened after the session became idle, skipping marking as idle")
-			idleSince = metav1.NewTime(rt)
-		} else if err != nil {
```

Comment:
Here was the problem. The idle flag is set to true. But then in the if statement below is was supposed to be set to false (and the logs say so) but the flag is never set to false.